### DISCUSSION TITLE
Is there an equivalent for arrayUnion or arrayRemove in Supabase?

### DISCUSSION THREAD

**Discussion by cdedreuille** [supabase/supabase#753] [Questions]
Hi,

I was wondering if there was an easy way to update an array in Supabase avoiding any duplications. In firebase I'm using `FieldValue.arrayUnion(element)`. Is there an equivalent in Supabase or an alternative way to do this?

**Comment by steve-chavez** [comment]
There's no   `array_union`, but you could create a similar SQL function:

```sql
create or replace function extensions.array_union(a1 anyarray, a2 anyarray) returns anyarray as $$
select array_agg(arr order by arr)
from (
    select distinct unnest(a1 || a2) as arr
) _;
$$ language sql;
```

Then this will work for any type of array(try it on the SQL Editor):

```sql
select array_union(ARRAY[1,2,3], ARRAY[2,3,4,5,6]);
[1,2,3,4,5,6]

select array_union(ARRAY['a','b','c'], ARRAY['b','c','d','e']);
["a","b","c","d","e"]
```

For `arrayRemove`, there is a builtin [array_remove](https://www.postgresql.org/docs/current/functions-array.html#id-1.5.8.25.6.2.2.11.1.1.1) that you can use like:

```sql
select array_remove(ARRAY[1,2,3], 2);
[1,3]
```

**Reply by cdedreuille** [reply]
Thanks a lot for the detailed answer. Sorry, I should have given more precision as I was looking to do this on the client or a serverless function. I ended up using `_.union()` from Lodash but this required that I have the previous data. Not ideal but it works for now.
